utils/fs/memory/memory.go

```diff
@@ -298,12 +281,7 @@ type content struct {
 }
 
 func (c *content) WriteAt(p []byte, off int64) (int, error) {
```

Comment:
There is no need for grow or copy here, just use the append function provided by the language. And remove size, this is what the length of a slice is for. You are rewriting here all the internal code of a slice.